test/controllers/propertyDetails/PropertyDetailsDateOfRevalueControllerSpec.scala

```diff
@@ -149,6 +151,58 @@ class PropertyDetailsDateOfRevalueControllerSpec extends PlaySpec with GuiceOneS
       }
     }
 
+    "collect information from cache and save to database" when {
+      "newRevaluedFeature flag is set to true and user clicks button 'save and continue'" in new Setup {
+        val inputJson: JsValue = Json.obj(
+          "dateOfRevalue" -> Json.obj(
+            "day" -> 1,
+            "month" -> 4,
+            "year" -> 2020
+          )
+        )
+        val authMock = authResultDefault(AffinityGroup.Organisation, defaultEnrolmentSet)
+        setAuthMocks(authMock)
+
+        when(mockAppConfig.newRevaluedFeature).thenReturn(true)
+        when(mockServiceInfoService.getPartial(ArgumentMatchers.any(), ArgumentMatchers.any(), ArgumentMatchers.any())).thenReturn(Future.successful(customBtaNavigationLinks))
+        when(mockDataCacheConnector.fetchAtedRefData[String](ArgumentMatchers.eq(DelegatedClientAtedRefNumber))(ArgumentMatchers.any(), ArgumentMatchers.any())).thenReturn(Future.successful(Some("XN1200000100001")))
+        when(mockDataCacheConnector.fetchAndGetFormData[Boolean](ArgumentMatchers.any())
+          (ArgumentMatchers.any(), ArgumentMatchers.any())).thenReturn(Future.successful(None))
+        when(mockBackLinkCacheConnector.saveBackLink(ArgumentMatchers.any(), ArgumentMatchers.any())(ArgumentMatchers.any())).thenReturn(Future.successful(None))
+
+        when(mockDataCacheConnector.fetchAndGetFormData[HasBeenRevalued](ArgumentMatchers.eq(HasPropertyBeenRevalued))(ArgumentMatchers.any(), ArgumentMatchers.any())).thenReturn(Future.successful(Some(HasBeenRevalued(Some(true)))))
+        when(mockDataCacheConnector.fetchAndGetFormData[PropertyDetailsNewValuation](ArgumentMatchers.eq(propertyDetailsNewValuationValue))
+          (ArgumentMatchers.any(), ArgumentMatchers.any()))
+          .thenReturn(Future.successful(Some(PropertyDetailsNewValuation(Some(1000000)))))
+        when(mockDataCacheConnector.fetchAndGetFormData[DateOfChange](ArgumentMatchers.eq(FortyThousandValueDateOfChange))
+          (ArgumentMatchers.any(), ArgumentMatchers.any()))
+          .thenReturn(Future.successful(Some(DateOfChange(Some(LocalDate.of(2021, 6, 15))))))
```

Comment:
The date matches the expected values as shown below: partAcqDispDate = Some(LocalDate.of(2021, 6, 15))

The reason for using the name 'partAcqDispDate' is that it corresponds to the existing naming convention in the PropertyDetailsRevalued model:

case class PropertyDetailsRevalued(
    isPropertyRevalued: Option[Boolean] = None,
    revaluedValue: Option[BigDecimal] = None,
    revaluedDate: Option[LocalDate] = None,
    partAcqDispDate: Option[LocalDate] = None
)
Since this model is used when the feature flag is off, I preferred not to change the names of the values within it to maintain consistency. Additionally, as this is the same model you suggested for saving the data, I proceeded with it accordingly.